--- /tmp/pygame-1.8.1release/src/rwobject.c	2008-07-22 07:32:57.000000000 -0400
+++ rwobject.c	2008-10-12 23:33:42.000000000 -0400
@@ -444,16 +444,16 @@
         Py_XDECREF (result);
     }
 
+    PyThreadState_Swap (oldstate);
+    PyThreadState_Clear (helper->thread);
+    PyThreadState_Delete (helper->thread);
+
     Py_XDECREF (helper->seek);
     Py_XDECREF (helper->tell);
     Py_XDECREF (helper->write);
     Py_XDECREF (helper->read);
     Py_XDECREF (helper->close);
 
-    PyThreadState_Swap (oldstate);
-    PyThreadState_Clear (helper->thread);
-    PyThreadState_Delete (helper->thread);
-
     PyMem_Del (helper);
 
     PyEval_ReleaseLock ();
