--- src/audio/SDL_audio.old.c	2009-10-06 00:12:01.000000000 +0000
+++ src/audio/SDL_audio.c	2009-10-06 00:14:10.000000000 +0000
@@ -39,15 +39,15 @@
 #if SDL_AUDIO_DRIVER_BSD
 	&BSD_AUDIO_bootstrap,
 #endif
-#if SDL_AUDIO_DRIVER_OSS
-	&DSP_bootstrap,
-	&DMA_bootstrap,
+#if SDL_AUDIO_DRIVER_PULSE
+	&PULSE_bootstrap,
 #endif
 #if SDL_AUDIO_DRIVER_ALSA
 	&ALSA_bootstrap,
 #endif
-#if SDL_AUDIO_DRIVER_PULSE
-	&PULSE_bootstrap,
+#if SDL_AUDIO_DRIVER_OSS
+	&DSP_bootstrap,
+	&DMA_bootstrap,
 #endif
 #if SDL_AUDIO_DRIVER_QNXNTO
 	&QNXNTOAUDIO_bootstrap,
