diff -ru libav-9.6.old/libavutil/file.c libav-9.6/libavutil/file.c
--- libav-9.6.old/libavutil/file.c	2013-05-12 03:11:15.000000000 -0400
+++ libav-9.6/libavutil/file.c	2013-06-05 21:19:50.351029580 -0400
@@ -50,7 +50,7 @@
 {
     FileLogContext file_log_ctx = { &file_log_ctx_class, log_offset, log_ctx };
     int err, fd = open(filename, O_RDONLY);
-    struct stat st;
+    struct _stati64 st;
     av_unused void *ptr;
     off_t off_size;
     char errbuf[128];
@@ -63,7 +63,7 @@
         return err;
     }
 
-    if (fstat(fd, &st) < 0) {
+    if (_fstati64(fd, &st) < 0) {
         err = AVERROR(errno);
         av_strerror(err, errbuf, sizeof(errbuf));
         av_log(&file_log_ctx, AV_LOG_ERROR, "Error occurred in fstat(): %s\n", errbuf);
