--- src/surface.c.old	2013-04-04 02:56:16.490683908 -0400
+++ src/surface.c	2013-04-04 02:54:51.411930744 -0400
@@ -2266,9 +2266,9 @@
         /* special case, SDL works */
         (dst->format->BytesPerPixel == 2 || dst->format->BytesPerPixel == 4))
     {
-	/* Py_BEGIN_ALLOW_THREADS */
+	Py_BEGIN_ALLOW_THREADS
         result = pygame_AlphaBlit (src, srcrect, dst, dstrect, the_args);
-	/* Py_END_ALLOW_THREADS */
+	Py_END_ALLOW_THREADS
     }
     else if (the_args != 0 ||
 	     (src->flags & (SDL_SRCALPHA | SDL_SRCCOLORKEY) &&
@@ -2280,15 +2280,15 @@
 	      dst->pixels == src->pixels &&
 	      surface_do_overlap (src, srcrect, dst, dstrect)))
     {
-	/* Py_BEGIN_ALLOW_THREADS */
+	Py_BEGIN_ALLOW_THREADS
         result = pygame_Blit (src, srcrect, dst, dstrect, the_args);
-	/* Py_END_ALLOW_THREADS */
+        Py_END_ALLOW_THREADS
     }
     /* can't blit alpha to 8bit, crashes SDL */
     else if (dst->format->BytesPerPixel == 1 &&
 	     (src->format->Amask || src->flags & SDL_SRCALPHA))
     {
-	/* Py_BEGIN_ALLOW_THREADS */
+	Py_BEGIN_ALLOW_THREADS
 	if (src->format->BytesPerPixel == 1)
 	{
 	    result = pygame_Blit (src, srcrect, dst, dstrect, 0);
@@ -2306,13 +2306,13 @@
 		result = -1;
 	    }
 	}
-	/* Py_END_ALLOW_THREADS */
+	Py_END_ALLOW_THREADS
     }
     else
     {
-	/* Py_BEGIN_ALLOW_THREADS */
+	Py_BEGIN_ALLOW_THREADS
         result = SDL_BlitSurface (src, srcrect, dst, dstrect);
-	/* Py_END_ALLOW_THREADS */
+	Py_END_ALLOW_THREADS
     }
 
     if (subsurface)
