diff -U 5 -r SDL2-2.0.4.orig/src/video/windows/SDL_windowsmodes.c SDL2-2.0.4/src/video/windows/SDL_windowsmodes.c
--- SDL2-2.0.4.orig/src/video/windows/SDL_windowsmodes.c	2016-01-02 14:56:31.000000000 -0500
+++ SDL2-2.0.4/src/video/windows/SDL_windowsmodes.c	2016-02-10 23:14:04.518428388 -0500
@@ -107,18 +107,21 @@
     if (index == ENUM_CURRENT_SETTINGS
         && (hdc = CreateDC(deviceName, NULL, NULL, NULL)) != NULL) {
         char bmi_data[sizeof(BITMAPINFOHEADER) + 256 * sizeof(RGBQUAD)];
         LPBITMAPINFO bmi;
         HBITMAP hbm;
+
+#if 0
         int logical_width = GetDeviceCaps( hdc, HORZRES );
         int logical_height = GetDeviceCaps( hdc, VERTRES );
 
         data->ScaleX = (float)logical_width / devmode.dmPelsWidth;
         data->ScaleY = (float)logical_height / devmode.dmPelsHeight;
         mode->w = logical_width;
         mode->h = logical_height;
-
+#endif
+        
         // WIN_GetMonitorDPI needs mode->w and mode->h
         // so only call after those are set.
         if (vid_data->GetDpiForMonitor) {
             WIN_GetMonitorDPIData dpi_data;
             RECT monitor_rect;
