--- src/video/quartz/SDL_QuartzVideo.m	2012-01-19 01:30:06.000000000 -0500
+++ src/video/quartz/SDL_QuartzVideo.m.new	2014-07-22 22:19:47.214252373 -0400
@@ -1045,7 +1049,9 @@
             [ qz_window setFrameTopLeftPoint:NSMakePoint((float) origin_x, (float) (this->info.current_h - origin_y))];
             center_window = 0;
         } else if ( center_window ) {
-            [ qz_window center ];
+	  NSRect visible = [ [ qz_window screen ] visibleFrame ];
+
+	  [ qz_window setFrameOrigin: NSMakePoint(visible.origin.x + (visible.size.width - width) / 2, visible.origin.y + (visible.size.height - height) / 2) ];
         }

         [ qz_window setDelegate:
