--- src/video/quartz/SDL_QuartzVideo.m.old	2013-05-19 23:27:49.508079063 -0400
+++ src/video/quartz/SDL_QuartzVideo.m	2013-05-19 23:29:04.230980253 -0400
@@ -573,6 +573,7 @@
         /* If we still have a valid window, close it. */
         if ( qz_window ) {
             NSCAssert([ qz_window delegate ] == nil, @"full screen window shouldn't have a delegate"); /* if that should ever change, we'd have to release it here */
+            [ NSApp removeWindowsItem: qz_window ];
             [ qz_window close ]; /* includes release because [qz_window isReleasedWhenClosed] */
             qz_window = nil;
             window_view = nil;
@@ -616,6 +617,7 @@
     /* Release window mode resources */
     else {
         id delegate = [ qz_window delegate ];
+        [ NSApp removeWindowsItem: qz_window ];
         [ qz_window close ]; /* includes release because [qz_window isReleasedWhenClosed] */
         if (delegate != nil) [ delegate release ];
         qz_window = nil;
